File: sum.c
#include <string.h>
#include "erl_nif.h"

int sum(int a, int b) {
  return a+b;
}

int factorial(int a){
  if(a == 0) return 1;
  else return a*factorial(a-1);
}

int string_length(char* string) {
  return strlen(string);
}

char * to_string(ErlNifBinary bin) {
  return bin.data;
}

File: sum.erl
-module(sum).
-export([sum/2, factorial/1, binary_length/1, to_string/1]).
-on_load(init/0).

init() ->
    ok = erlang:load_nif("./sum_nif", 2).

sum(_X, _Y) ->
    exit(nif_library_not_loaded).

string_length(_Y) ->
    exit(nif_library_not_loaded).

binary_length(_Y) ->
    exit(nif_library_not_loaded).

factorial(_) ->
    exit(nif_library_not_loaded).

to_string(_) ->
    exit(nif_library_not_loaded).
